spec/controllers/agents/dry_runs_controller_spec.rb

```diff
@@ -100,5 +100,22 @@ def valid_attributes(options = {})
       results = assigns(:results)
       expect(results[:log]).to match(/^\[\d\d:\d\d:\d\d\] INFO -- : Fetching #{Regexp.quote(url_from_event)}$/)
     end
+
+    it "uses the memory of an existing Agent" do
+      valid_params = {
+        :name => "somename",
+        :options => {
+          :code => "Agent.check = function() { this.createEvent({ 'message': this.memory('fu') }); };",
+        }
+      }
+      agent = Agents::JavaScriptAgent.new(valid_params)
```

Comment:
Something minor, you could do `agent = Agent.build_for_type(Agents::JavaScriptAgent, users(:bob), memory: {fu: 'bar'})` to save a few lines.